src/foundation.cs

```diff
@@ -12226,13 +13150,22 @@ interface NSUnitTemperature : NSSecureCoding {
 		NSDimension BaseUnit { get; }
 	}
 
-#if MONOMAC
+#if !WATCH && !TVOS
+	delegate void FileProviderMessageInterfacesCompletionHandler (NSFileProviderMessageInterface [] interfaces, NSError error);
+
+	[Mac (10,8), iOS (11,0), NoWatch, NoTV]
 	partial interface NSFileManager {
 
 		[MountainLion, Export ("trashItemAtURL:resultingItemURL:error:")]
 		bool TrashItem (NSUrl url, out NSUrl resultingItemUrl, out NSError error);
+
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		[Async, Export ("getFileProviderMessageInterfacesForItemAtURL:completionHandler:")]
+		void GetFileProviderMessageInterfaces (NSUrl url, FileProviderMessageInterfacesCompletionHandler completionHandler);
```

Comment:
Does `FileProviderMessageInterfacesCompletionHandler` exists in ObjC ? (it's not prefixed by `NS`)

When we have a single parameter *or* a two parameters with a `NSError` then we can (when it's not confusing) use `Action<T[,V]>`. That makes it easier to share delegates in code and reduce metadata (app size).

The reason why we can't do it for longer delegates is because the arguments become unclear. That's not the case here since the method name is clear about what will be given and the `NSError` itself is self-documenting.